### PR TITLE
NIFI-9628: Added a uiOnly flag when requesting Controller Service det…

### PR DESCRIPTION
**Pull request by markap14**
…ails and the list of Controller Services. This allows us to return much less data when retrieving these resources.

Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

_Enables X functionality; fixes bug NIFI-YYYY._

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [ ] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-process-group-configuration.js [hunk 1]
```diff
@@ -348,7 +354,10 @@
         var parameterContexts = $.ajax({
                 type: 'GET',
                 url: config.urls.parameterContexts,
-                dataType: 'json'
+                dataType: 'json',
+                data: {
+                    uiOnly: true
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-process-group-configuration.js:359]: The parameter contexts endpoint does not support the `uiOnly` parameter. Was this added inadvertently?

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/util/SnippetUtils.java [hunk 2]
```diff
@@ -1015,4 +1018,33 @@ private void normalizeCoordinates(Collection<? extends ComponentDTO> components)
         }
     }
 
+    public static void stripNonUiRelevantFields(final ControllerServiceEntity serviceEntity) {
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/util/SnippetUtils.java:1021]: What was the motivation for adding these methods to `SnippetUtils`?

**Review by mcgilman — COMMENTED** [review]
Thanks for the PR @markap14! Considering the motivation for this PR, I think you could also consider adding the `uiOnly` flag to `PUT /nifi-api/controller-services/{id}/references` and `PUT /nifi-api/controller-services/{id}/run-status`. Currently when enabling or disabling a Controller Service we invoke these APIs and it is still returning the referencing component `descriptors` and `properties`.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/util/SnippetUtils.java (hunk 2 above)
Comment (markap14) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/util/SnippetUtils.java:1021]: It's referenced from a few different places. So I wanted to extract the logic into some sort of utility class that could be easily referenced from all of the resources that needed it. We could create a separate util class  - `ControllerServiceUtils` or something like that maybe? Didn't feel like that warranted its own util class, and i figured one could argue that this deals with a snippet of the flow. Though I suppose this util class is actually intended for interacting with Snippets with a capital S, the specific components. I could move it into its own util class if you prefer.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-process-group-configuration.js (hunk 1 above)
Comment (markap14) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-process-group-configuration.js:359]: Oops, good catch. That was definitely inadvertently

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/util/SnippetUtils.java (hunk 2 above)
Comment (mcgilman) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/util/SnippetUtils.java:1021]: `SnippetUtils` is intended for interacting with Snippets with a capital S. We could do another class or we could even put it in the `ApplicationResource`. I believe all the calling code is in a class that extends `ApplicationResource`.

Comment (markap14) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/util/SnippetUtils.java:1021]: Yeah fair. Could definitely put it in ApplicationResource.

**Comment by markap14** [comment]
Thanks for reviewing @mcgilman . I pushed a new commit that I believe addresses all of the above.

**Review by mcgilman — APPROVED** [review]
@markap14 thanks for the updates! +1 looks great.

**Pull request merged**